### DISCUSSION TITLE
help with optimizing config

### DISCUSSION THREAD

**Discussion by ckokt** [swar/Swar-Chia-Plot-Manager#561] [Q&A] [answered]
Equipments:
CPU: Ryzen 9-5950x - 16core
RAM: 64gb
NVME: 2tb *3
HDD: 6tb *2
Motherboard: Gigabyte X570 AORUS MASTER

Thank you.
Best Regards.

**Comment by Lyushen** [comment, marked as answer]
1. Windows user? No raid in that case. 3 separate NVMe with 3 tasks perform 3x times better.
3. RAM 4k and it does not metter.
4. I believe you cannot utilize 2TB m2 because of limitation of its speed and IOPS. Try it, mine can handle 4 tasks per one NVMe, but I limited by 1TB. Its highly depends on controller and how its handle long term usage. Look [here](https://www.tomshardware.com/reviews/adata-xpg-gammix-s50-lite-m-2-ssd-review/2) and pay attention to this [graph ](https://cdn.mos.cms.futurecdn.net/Trt5oYZaJRiekwMpwoLaCS-970-80.png.webp). I chose 980 Pro instead of TBW value. So it's not bottlenecking my 10/20@5GHz CPU.

5. Try this config, you need to figure out the best "threads:" and "max_for_phase_1" (6 is more than enough I believe) within "max_concurrent" should be **Threads** _x_ **max_for_phase_1** _x_ **3x (NVMe)** not more than your CPU's 28 threads (~4-6 threads for the rest other background phases). Play with it.

```
manager:
check_interval: 60
log_level: ERROR

global:
max_concurrent: 12

jobs:
- name: NVMe 1
max_plots: 999
farmer_public_key:
pool_public_key:
temporary_directory: D:\
temporary2_directory:
destination_directory: G:\
size: 32
bitfield: true
threads: 3
buckets: 128
memory_buffer: 4000
max_concurrent: 4
max_concurrent_with_start_early: 4
stagger_minutes: 60
max_for_phase_1: 2
concurrency_start_early_phase: 4
concurrency_start_early_phase_delay: 30
temporary2_destination_sync: false

- name: NVMe 2
max_plots: 999
farmer_public_key:
pool_public_key:
temporary_directory: F:\
temporary2_directory:
destination_directory: G:\
size: 32
bitfield: true
threads: 4
buckets: 128
memory_buffer: 4000
max_concurrent: 4
max_concurrent_with_start_early: 4
stagger_minutes: 60
max_for_phase_1: 2
concurrency_start_early_phase: 4
concurrency_start_early_phase_delay: 30
temporary2_destination_sync: false

- name: NVMe 3
max_plots: 999
farmer_public_key:
pool_public_key:
temporary_directory: E:\
temporary2_directory:
destination_directory: G:\
size: 32
bitfield: true
threads: 5
buckets: 128
memory_buffer: 4000
max_concurrent: 4
max_concurrent_with_start_early: 4
stagger_minutes: 60
max_for_phase_1: 2
concurrency_start_early_phase: 4
concurrency_start_early_phase_delay: 30
temporary2_destination_sync: false
```

**Reply by ckokt** [reply]
i have problem on new version, why it start after 2 hours?

![swar config](https://user-images.githubusercontent.com/2990501/119119003-58d4d480-ba5d-11eb-9db5-ef696d37ff5d.jpg)

this is my setting :

  max_concurrent: 16
  max_for_phase_1: 3
  minimum_minutes_between_jobs: 5

  - name: p1
    max_plots: 999
    farmer_public_key:
    pool_public_key:
    temporary_directory: D:\
    temporary2_directory:
    destination_directory: G:\
    size: 32
    bitfield: true
    threads: 3
    buckets: 128
    memory_buffer: 4000
    max_concurrent: 6
    max_concurrent_with_start_early: 6
    initial_delay_minutes: 0
    stagger_minutes: 60
    max_for_phase_1: 2
    concurrency_start_early_phase: 4
    concurrency_start_early_phase_delay: 30
    temporary2_destination_sync: false
    exclude_final_directory: false
    skip_full_destinations: true
    unix_process_priority: 10
    windows_process_priority: 32
    enable_cpu_affinity: false
    cpu_affinity: [ 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31 ]

  - name: p2
    max_plots: 999
    farmer_public_key:
    pool_public_key:
    temporary_directory: E:\
    temporary2_directory:
    destination_directory: G:\
    size: 32
    bitfield: true
    threads: 4
    buckets: 128
    memory_buffer: 4000
    max_concurrent: 6
    max_concurrent_with_start_early: 6
    initial_delay_minutes: 0
    stagger_minutes: 60
    max_for_phase_1: 2
    concurrency_start_early_phase: 4
    concurrency_start_early_phase_delay: 30
    temporary2_destination_sync: false
    exclude_final_directory: false
    skip_full_destinations: true
    unix_process_priority: 10
    windows_process_priority: 32
    enable_cpu_affinity: false
    cpu_affinity: [ 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31 ]

  - name: p3
    max_plots: 999
    farmer_public_key:
    pool_public_key:
    temporary_directory: F:\
    temporary2_directory:
    destination_directory: H:\
    size: 32
    bitfield: true
    threads: 5
    buckets: 128
    memory_buffer: 4000
    max_concurrent: 6
    max_concurrent_with_start_early: 6
    initial_delay_minutes: 0
    stagger_minutes: 60
    max_for_phase_1: 2
    concurrency_start_early_phase: 4
    concurrency_start_early_phase_delay: 30
    temporary2_destination_sync: false
    exclude_final_directory: false
    skip_full_destinations: true
    unix_process_priority: 10
    windows_process_priority: 32
    enable_cpu_affinity: false
    cpu_affinity: [ 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31 ]

**Comment by ckokt** [comment]
Thanks alot!!! Thanks for your big help!! 👍👍👍
